mediaapi/routing/download.go

```diff
@@ -673,6 +675,36 @@ func (r *downloadRequest) fetchRemoteFileAndStoreMetadata(
 	return nil
 }
 
+func (r *downloadRequest) GetContentLengthAndReader(contentLengthHeader string, body *io.ReadCloser) (int64, io.Reader, error) {
+	var reader io.Reader
+	var contentLength int64
+
+	if contentLengthHeader != "" {
+		parsedLength, parseErr := strconv.ParseInt(contentLengthHeader, 10, 64)
+
+		if parseErr != nil {
+			r.Logger.WithError(parseErr).Warn("Failed to parse content length")
+			return 0, nil, errors.Wrap(parseErr, "invalid response from remote server")
+		}
+
+		reader = *body
+		contentLength = parsedLength
+	} else {
+		// Content-Length header is missing, we need to read the whole body to get its length
+		bodyBytes, readAllErr := ioutil.ReadAll(*body)
```

Comment:
This is going to be a problem if the file is huge as it'll all be put into memory. Why do you need to do this? I _think_ the previous solution didn't do this but because you force pushed I can't cross-reference with what was previously done.

Reply:
Suggestion: return 0 by default. Final size is set using `bytesWritten` from `hash, bytesWritten, tmpDir, err := fileutils.WriteTempFile(ctx, reader, maxFileSizeBytes, absBasePath)` anyway.